docs/pages/integrations/django/index.md

```diff
@@ -129,6 +129,49 @@ Class-based views are also supported. Specify dependencies in the class `__init_
 
 For more examples, see the [Wireup Django integration tests](https://github.com/maldoinc/wireup/tree/master/test/integration/django/view.py).
 
+#### Non core-django views (eg, Django REST framework)
+If your project uses third-party packages to create views, such as [Django REST framework](https://www.django-rest-framework.org/), you must use the `@inject` decorator explicitly.
+
+```python title="app/views.py"
+from rest_framework.decorators import api_view
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import ViewSet
+
+from wireup import Injected
+from wireup.integration.django import inject
+
+from mysite.polls.services import S3Manager
+
+
+@api_view(("GET",))
+@inject
+def drf_function_based_view(request: Request, s3_manager: Injected[S3Manager]) -> Response:
+    # Use the injected S3Manager instance
+    return Response(...)
+
+
+class DRFClassBasedView(APIView):
+    @inject
+    def get(self, request: Request, s3_manager: Injected[S3Manager]) -> Response:
+        # Use the injected S3Manager instance
+        return Response(...)
+
+
+class DRFViewSet(ViewSet):
+    @inject
+    def list(self, request: Request, greeter: Injected[GreeterService]) -> Response:
+        # Use the injected S3Manager instance
+        return Response(...)
+```
+
+If your project shares core and non core-django views it's strongly recommended to use `@inject` across all your views for consistency.
+
+!!! warning "Don't mix injection approaches in Django class-based views"
+    When using `@inject` on methods, do not also use `Injected[]` annotations in `__init__`. 
+    Pick one approach: either use `@inject` on methods, or rely on auto-injection for `__init__` parameters.
```

Comment:
We can just reword this as a suggestion and make it a "tip"/"best practice" type rather than "warning". We can also apply tabs to show a do/don't for this similar to https://maldoinc.github.io/wireup/latest/tips_tricks/.

To consider: There is also one huge benefit to not marking core django views with `@inject` though: Wireup will check and validate all injections which is not possible when using the "inject unchecked" function.